Address @r12a's comments plus add spillover definition

aphillips opened the pull request:
<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/i18n-glossary/pull/27.html" title="Last updated on Mar 27, 2023, 7:59 PM UTC (b5ab234)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/i18n-glossary/27/43c693b...aphillips:b5ab234.html" title="Last updated on Mar 27, 2023, 7:59 PM UTC (b5ab234)">Diff</a>

aphillips committed:
Address @r12a's comments.

aphillips committed:
Merge branch 'w3c:gh-pages' into gh-pages

aphillips committed:
Addresses #30: addition of spillover definition

r12a commented on index.html:
```diff
@@ -548,13 +550,15 @@ <h4>Relationship to the Infra Standard</h4>
 </pre>
 
-    <p>This allows you to then write references using the normal ReSpec notation (<code>&lt;a&gt;</code> tags or <code>[&#x200b;=term=&#x200b;]</code> markup).</p>
+    <p>Adding the above configuration allows you to write references the terms found here using the normal ReSpec notation (<code>&lt;a&gt;</code> tags or <code>[&#x200b;=term=&#x200b;]</code> markup).</p>
```
> to write references the terms found here using

became a little scrambled

aphillips replied:
Done.

r12a commented on index.html:
```diff
@@ -427,6 +427,8 @@ <h2>Glossary</h2>
 
 <p><dfn class="lint-ignore export">Spacing mark</dfn>. Combining characters that consume space along the baseline, either before or after the base character.</p>
 
+<p><dfn class="link-ignore export">Spillover</dfn>. Errors in text presentation due to improper [=bidi isolation=]. When one string is inserted into another string, unless the inserted value is isolated and given the proper [=base direction=], the [=bidi algorithm=] can visually rearrange the text in unexpected ways that make the text illegible. For example, a MessageFormat pattern string like <c ...
```
due to improper [=bidi isolation=] -> due to a lack of [=bidi isolation=]

i don't think you need to show the (lengthy) source code: just show what it should look like and what it turns out as.  That would enable you to show a second example based on list ordering.

r12a replied:
We should point to https://www.w3.org/International/articles/inline-bidi-markup/index.en.html for more information.

aphillips replied:
Done

aphillips committed:
Address teleconference comments.
- remove spillover example in favor of bidi isolation
- add right-to-left and left-to-right to RTL and LTR dfns
- Tweak wording of spillover
- Link bidi isolation to spillover

aphillips committed:
Add link to bidi article per @r12a's suggestion

aphillips committed:
Address comment about respec instruction.

aphillips committed:
Add 'bidi isolate' and update 'bidi isolation' to read like a definition

The current definition of bidi isolation is not written like a
definition. Making it into a definition was easier by introducing the
concept of a bidi isolate and then referring to it.

aphillips commented:
@r12a I added 822b7b7 because the term `bidi isolation` wasn't written like a definition. Please have a look at that part of the change and make comments.

aphillips committed:
Fix the 'H' head (it was lowercase)

r12a commented:
I finally found some time to look at this.  I have a couple of concerns that i'll raise as inline comments.

r12a commented on index.html:
```diff
@@ -164,7 +164,9 @@ <h2>Glossary</h2>
 <p><dfn data-lt="basic language range|Basic language range" class="lint-ignore export">Basic language range</dfn>. A [=language range=] consisting of a sequence of subtags separated by hyphens. That is, it is identical in appearance to a language tag.</p>
 
-<p><dfn class="lint-ignore export">Bidi isolation</dfn>. Isolation often needs to be applied to a range of text in order to prevent the automatic rules of the Unicode Bidirectional Algorithm incorrectly ordering that content in relation to the surrounding text. For example, numbers following right-to-left text in memory are automatically positioned to the left of that text by the Bidi Algorithm,  ...
+<p><dfn class="lint-ignore export" data-lt="bidi isolate">Bidirectional isolate</dfn>. Defined by the [=Unicode Bidirectional Algorithm=] [[UAX9]], an <em>isolate</em> is a range of text whose contents do not participate in the embedding and reordering of the surrounding text the range appears in, which prevents the bidi algorithm from incorrectly ordering that content in relation to the surround ...
```
I think this is a tricky definition to pin down, and is one we're inventing rather than something explicitly _defined_ in UAX#9.

In UAX#9 the concept of 'isolate' appears to be defined in terms of text that is bounded by specific formatting characters, namely LRI, RLI, or FSI.  I don't see any 'definition' of isolate – only a description of isolate formatting characters that in only a small handful of places refers to the term isolate when referring to the text itself.

'Isolates' as described here do in fact participate in the reordering of surrounding text, but do so as a neutral unit.

Also, the sentence is very long.

Suggestion:
A range of text, bounded by formatting characters or markup, such that the contents do not interact with the surrounding text per the rules of the  [=Unicode Bidirectional Algorithm=] [[UAX9]]. In certain circumstances, isolation prevents the  incorrect ordering of content by the bidi algorithm.

aphillips replied:
Agreed about the complexity here. 

I like the suggestion okay, but the "certain circumstances" part made me nervous. I went back to UAX9 and harvested text found in https://www.unicode.org/reports/tr9/#Explicit_Directional_Isolates to make this:

```suggestion
<p><dfn class="lint-ignore export" data-lt="bidi isolate">Bidirectional isolate</dfn>. A range of text, bounded by formatting characters or markup, that is treated by the [=Unicode Bidirectional Algorithm=] [[UAX9]] as directionally isolated from its surroundings. The entire range of text inside the isolate is treated by the surrounding text as if it were a single neutral character (such as <span class="codepoint" translate="no"><bdi lang="en">&#xFFFC;</bdi> [<span class="uname">U+FFFC OBJECT REPLACEMENT CHARACTER</span>])</span>, and is assigned the corresponding display position in the surrounding text. Furthermore, the text inside the isolate has no effect on the ordering of the text outside it, and vice versa.</p>
```

See #34 for this as part of a PR. Please comment on it!

r12a replied:
I think this has become a tutorial rather than a definition.

aphillips replied:
Fair enough, although it is quoting UAX9.

r12a commented on index.html:
```diff
@@ -164,7 +164,9 @@ <h2>Glossary</h2>
-<p><dfn class="lint-ignore export">Bidi isolation</dfn>. Isolation often needs to be applied to a range of text in order to prevent the automatic rules of the Unicode Bidirectional Algorithm incorrectly ordering that content in relation to the surrounding text. For example, numbers following right-to-left text in memory are automatically positioned to the left of that text by the Bidi Algorithm,  ...
+<p><dfn class="lint-ignore export" data-lt="bidi isolate">Bidirectional isolate</dfn>. Defined by the [=Unicode Bidirectional Algorithm=] [[UAX9]], an <em>isolate</em> is a range of text whose contents do not participate in the embedding and reordering of the surrounding text the range appears in, which prevents the bidi algorithm from incorrectly ordering that content in relation to the surround ...
+
+<p><dfn class="lint-ignore export">Bidi isolation</dfn>. The use of [=bidi isolates=] in text in order to prevent the automatic rules of the Unicode Bidirectional Algorithm incorrectly ordering that content in relation to the surrounding text. For example, numbers following right-to-left text in memory are automatically positioned to the left of that text by the Bidi Algorithm, but sometimes need ...
```
positioned to the left of that text -> positioned to the left of RTL text

lists of RTL items occur in a LTR sentence -> a list of RTL items occur in a LTR sentence (and vice versa)

r12a replied:
occur -> occurs

aphillips replied:
Done (in #34)

r12a replied:
Not quite. 
lists of RTL items occurs -> a list of RTL items occurs

xfq commented on index.html:
```diff
@@ -532,14 +536,16 @@ <h2>Glossary</h2>
-	<p>If you are writing a W3C specification, you can use the terminology in this document directly by importing this document as a cross-reference source. This may be especially helpful when using <a>character encoding</a> or <a>locale</a>-related terms found here. Such terms are common to the needs of many specifications and often refer to other definitions found in this glossary. In most cases,  ...
+	<p>If you are writing a W3C specification, you can use the terminology in this document directly by importing this document as a cross-reference source. This may be especially helpful when using <a>character encoding</a> or <a>locale</a>-related terms found here. Such terms are common to the needs of many specifications and often refer to other definitions found in this glossary.</p>
+	
+	<p>This glossary is a Working Group NOTE, which means that the definitions found in it are, by definition, non-normative. In most cases, most specifications do not need the definition of terms found here to be formally normative. Using the definitions found here lends clarity when using specialized terminology, but doesn't affect requirements directly and in those cases copying these definitions ...
```
According to the [MOS](https://www.w3.org/Guide/manual-of-style/#Terms), 'Note' should be used instead of 'NOTE'.
